pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1339,7 +1339,32 @@ func (c *Cloud) InstanceExistsByProviderID(ctx context.Context, providerID strin
 
 // InstanceShutdownByProviderID returns true if the instance is in safe state to detach volumes
 func (c *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
-	return false, cloudprovider.NotImplemented
+	instanceID, err := kubernetesInstanceID(providerID).mapToAWSInstanceID()
+	if err != nil {
+		return false, err
+	}
+
+	request := &ec2.DescribeInstancesInput{
+		InstanceIds: []*string{instanceID.awsString()},
+	}
+
+	instances, err := c.ec2.DescribeInstances(request)
+	if err != nil {
+		return false, err
+	}
+	if len(instances) == 0 {
+		return false, nil
```

Comment:
Does this imply that we took so long that AWS removed the instance from the list?  I agree it is unexpected, but maybe `glog.Warning` and `return true, nil`

Reply:
+1 we should return true here with a warning

Reply:
If we return true here it means that instance is not removed from cluster. Instead shutdown taint is added. But the instance is deleted for real from aws and imo correct one is return false here. By returning false code will continue checks and finally remove node from cluster